### PR TITLE
fix(lsp): don't apply preload limit to workspace walk

### LINKED ISSUE #23121: Import map entries are not resolved in the editor
## Reproduction

Project structure

```
.vscode/
    settings.json
main.ts
deno.json
```

Filename: .vscode/settings.json

```json
{
  "deno.enable": true
}
```

Filename: main.ts

```ts
import "$std/dotenv/load.ts";
```

Filename: deno.json

```json
{
  "imports": {
    "$std/": "https://deno.land/std@0.216.0/"
  }
}
```

Open VS Code in the project's directory, and see if there is an error on the $std import.

## Notes

- Reproduces since 1.42.0, can’t reproduce with 1.41.3.
- Doesn’t matter if the file name is deno.jsonc or even if a standalone import_map.json is used.
- The code runs properly, and `deno check` passes on main.ts.
- OS: Fedora 39
- VS Code version: 1.87.2
- Deno extension version: 3.36.0

### LINKED ISSUE #23141: `deno.importMap` setting not respected when there is a `deno.json`
I just upgraded to 1.42.0 and I'm seeing something similar (worked fine with 1.41.3)

Code:
```js
import {xu} from "xu";
```

Error: `Relative import path "xu" not prefixed with / or ./ or ../`

I have an importMap:
```json
{
    "imports" :
    {
        "xu" : "/path/to/xu.js"
    }
}
```

In my VSCode  project.code-workspace:
```json
{
	"settings" : {
		"deno.enable"    : true,
		"deno.lint"      : true,
		"deno.unstable"  : true,
		"deno.cache"     : "/path/to/.deno/",
		"deno.importMap" : "/path/to/importMap.json",
		"deno.config": "/path/to/lintConfig.jsonc",
	}
}
```

Running from CLI works fine: `deno check --import-map /path/to/importMap.json myFile.js`

It's as if it's ignoring my deno.importMap directive entirely. Changing the value to a non-existant file, no errors or warnings show up in the `output` of `Deno Language Server` at all.

If I revert back to `deno 1.41.3` then I properly see in my `output`:
```
Using import map from workspace settings: "/path/to/importMap.json"
  Resolved import map: "file:///path/to/importMap.json"
```

This is entirely missing when running deno 1.42.0

_Originally posted by @Sembiance in https://github.com/denoland/deno/issues/23121#issuecomment-2028085581_

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes #23121.
Fixes #23141.

### CODE REVIEW

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -1069,8 +1069,7 @@ impl Inner {
 
   fn walk_workspace(config: &Config) -> (BTreeSet<ModuleSpecifier>, bool) {
     let mut workspace_files = Default::default();
-    let document_preload_limit =
-      config.workspace_settings().document_preload_limit;
+    let entry_limit = 1000;
```
Comment (dsherret) [inline cli/lsp/language_server.rs:1072]: I don't fully understand this, so I'll have to look at it in the morning, but I'm a bit concerned that a user would run into the same bug again once they have a lot of files.

Comment (nayeemrmn) [reply cli/lsp/language_server.rs:1072]: They'd need 1000 files at the top level preceding `deno.json` for that to happen. After the multi-deno.json detection is all set up, if there's demand I want to make some of it more lazy, like scanning upwards for new scopes whenever a file is opened in the editor. Let's leave that for later though.

**Comment by nayeemrmn** [comment]
Added fix for #23141.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**